crates/aptos-telemetry/src/service.rs

```diff
@@ -97,15 +94,16 @@ fn fetch_peer_id(node_config: &NodeConfig) -> String {
     }
 }
 
-async fn spawn_periodic_background_task<Fut>(interval_seconds: u64, task_fn: impl Fn() -> Fut)
-where
-    Fut: Future<Output = ()>,
-{
-    let mut interval = time::interval(Duration::from_secs(interval_seconds));
-    loop {
-        interval.tick().await;
-        task_fn().await;
-    }
+macro_rules! spawn_periodic_background_task {
+    ($i:expr, $fn:block) => {
+        tokio::spawn(async move {
+            let mut interval = time::interval(Duration::from_secs($i));
+            loop {
+                interval.tick().await;
+                $fn.await;
+            }
+        })
+    };
```

Comment:
noob trying to understand. why do we need to tokio spawn here?

Reply:
Async {} is a future in rust. To make progress it needs to be polled. Thats what tokio runtime is for. 
Now join will also poll futures but in the context of a single task. 
I changed this to spawn, because:
1. The ordering of events is less clear in the join scenario.
2. Any blocking event in any of these futures will block them all.
3. Its more Idiomatic and readable.

Reply:
Agree with 1 and 2, but in this case, we dont have any blocking events in any of the futures and they are all very thin and run periodically every 15 seconds minimum so we a single task is more than enough. We also renamed the method. About 3, it may not be idiomatic, but certainly keeps the code clean without the macro overhead.

Reply:
Thanks @Markuze ! I just discussed this internally with @ibalajiarun  . 
We'd prefer to keep the current future based code even if it's running in a single task since it's a bit easier to read than the macro based approach and we don't see a problem with blocking stuff since it's just doing cheap operations and async IO.
I renamed the utility function to `run_periodic_function` for clarity though.